### DISCUSSION TITLE
Dynamic imports

### DISCUSSION THREAD

**Discussion by dschewchenko** [ElMassimo/vite-plugin-image-presets#20] [Q&A]
Hello :)
Does this plugin supports vite's dynamic imports?
For example:
new URL(`/src/assets/img/category/${slug}.png`, import.meta.url).href

**Comment by ElMassimo** [comment]
Hi Dmytro!

I've only used this plugin with [Glob Imports](https://vitejs.dev/guide/features.html#glob-import).

I haven't tried the [single-level dynamic import](https://vitejs.dev/guide/features.html#dynamic-import), but it might work as well, as long as Vite is processing each possible asset.

**Reply by dschewchenko** [reply]
Thanks. Will check and report here :)

**Reply by dschewchenko** [reply]
It looks, like it works with glob import and passing query params as object in second argument

**Reply by dschewchenko** [reply]
with new URL() it doesn't work, because it returns only path, not object
